lib/redis.js

```diff
@@ -204,13 +204,16 @@ module.exports = CoreObject.extend({
       if (!revisions) {
         return;
       }
+      var promises = [];
       revisions.forEach(function(revision) {
         if (revision !== current) {
-          client.del(keyPrefix + ":" + revision);
-          client.del(keyPrefix + ":revision-data:" + revision);
-          client.zrem(listKey, revision);
+          promises.push(client.del(keyPrefix + ":" + revision));
+          promises.push(client.del(keyPrefix + ":revision-data:" + revision));
+          promises.push(client.zrem(listKey, revision));
         }
       });
+
+      return RSVP.all(promises);
```

Comment:
I was seeing flaky tests when we weren't waiting on these promises, so I wrapped it in an `RSVP.all` so we wait for all the redis operations to complete.